internal/debug/flags.go

```diff
@@ -141,30 +145,36 @@ func init() {
 // Setup initializes profiling and logging based on the CLI flags.
 // It should be called as early as possible in the program.
 func Setup(ctx *cli.Context) error {
-	logFile := ctx.String(logFileFlag.Name)
-	useColor := logFile == "" && os.Getenv("TERM") != "dumb" && (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd()))
+	output := io.Writer(os.Stderr)
 
-	var logfmt log.Format
+	var fmtr log.Format
 	if ctx.Bool(logjsonFlag.Name) {
-		logfmt = log.JSONFormat()
-	} else {
-		logfmt = log.TerminalFormat(useColor)
-	}
-
-	if logFile != "" {
-		var err error
-		logOutputStream, err = log.FileHandler(logFile, logfmt)
-		if err != nil {
-			return err
-		}
+		fmtr = log.JSONFormat()
 	} else {
-		output := io.Writer(os.Stderr)
-		if useColor {
+		usecolor := (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd())) && os.Getenv("TERM") != "dumb"
+		if usecolor {
 			output = colorable.NewColorableStderr()
 		}
-		logOutputStream = log.StreamHandler(output, logfmt)
+		fmtr = log.TerminalFormat(usecolor)
 	}
-	glogger.SetHandler(logOutputStream)
+
+	stdHandler := log.StreamHandler(output, fmtr)
+	logFolder := ctx.String(logLocationFlag.Name)
+	ostream := stdHandler
+	if err := validateLogLocation(logFolder); err != nil {
+		log.Warn("issue with initiatilizing file logger", "err", err)
+	} else if logFolder != "" {
+		lumberjackHandler := log.StreamHandler(&lumberjack.Logger{
+			Filename:   path.Join(logFolder, "log.log"),
+			MaxSize:    100,
+			MaxBackups: 7,
+			MaxAge:     10,
+		}, fmtr)
+
+		ostream = log.MultiHandler(lumberjackHandler, stdHandler)
+	}
```

Comment:
Might aswell abort the setup if the configured log location is unusable...? 

```suggestion
	if folder := ctx.String(logLocationFlag.Name); folder != "" {
		if err := validateLogLocation(folder); err != nil {
			return fmt.Errorf("failed to initiatilize file logger: %v", err)
		}
		lumberjackHandler := log.StreamHandler(&lumberjack.Logger{
			Filename:   path.Join(folder, "log.log"),
			MaxSize:    100,
			MaxBackups: 7,
			MaxAge:     10,
		}, fmtr)
		ostream = log.MultiHandler(lumberjackHandler, stdHandler)
	}
```

Reply:
This is _not_ resolved, you ignored the check for `folder != ""`. I think your code would now always use the folder-logger, even when none is specified.